File: portal/src/types/terminal.ts
import { TerminalStatus } from "./index";

export interface Terminal {
  id: string;
  tenant_id: string;
  name: string;
  description: string | null;
  api_key_prefix: string;
  location: string | null;
  status: TerminalStatus;
  last_seen_at: string | null;
  created_at: string;
}

export interface TerminalCreateRequest {
  name: string;
  description?: string;
  location?: string;
}

export interface TerminalCreateResponse extends Terminal {
  api_key: string;
}

export interface TerminalUpdateRequest {
  name?: string;
  description?: string;
  location?: string;
}

File: backend/app/schemas/terminal.py
from datetime import datetime

from pydantic import BaseModel


class TerminalCreate(BaseModel):
    name: str
    description: str | None = None
    location: str | None = None


class TerminalUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    location: str | None = None


class TerminalResponse(BaseModel):
    id: str
    tenant_id: str
    name: str
    description: str | None
    api_key_prefix: str
    location: str | None
    status: str
    last_seen_at: datetime | None
    created_at: datetime

    model_config = {"from_attributes": True}


class TerminalCreated(TerminalResponse):
    api_key: str


class TerminalListResponse(BaseModel):
    items: list[TerminalResponse]
    total: int
